src/components/elements/input/GenericSelect.tsx

```diff
@@ -41,18 +47,33 @@ const GenericSelect = <
         <TextInput
           {...params}
           {...inputProps}
-          InputProps={{ ...params.InputProps, ...inputProps.InputProps }}
+          InputProps={{
+            ...params.InputProps,
+            startAdornment: rest.loading ? (
+              <InputAdornment position='start'>
```

Comment:
We should only show the indicator if the selector has a value. Or else we are causing an unnecessary glitchy flash while the loading indicator shows up and then disappears.

```suggestion
            // Show a loading indicator if we have a value but the picklist is still loading
            startAdornment: rest.loading && params.inputProps.value ? (
              <InputAdornment position='start' sx={{ pl: 1 }}>
```

Reply:
fixed